### ISSUE TITLE
Placeholders {z}, {x}, and {y} Not Replaced When Adding RasterSource Programmatically

### ISSUE THREAD

**Issue by saevioapps** [maplibre/maplibre-native#2757]
**Describe the bug**
When adding a RasterSource programmatically to a MapLibreMap in Android, the placeholders {z}, {x}, and {y} in the tile URL are not being replaced with the correct zoom level and tile coordinates. This issue occurs even when using a simple map setup with an empty style and test tile source.

**To Reproduce**
    Initialize a MapView and load an empty style (fromJson("{}")).
    Programmatically add a RasterSource with a tile URL containing {z}, {x}, and {y} placeholders.
    Add a RasterLayer that uses the previously created RasterSource.

Here is the code to reproduce the issue:

```
AndroidView(factory = { mapView }, update = { mapView ->
    mapView.getMapAsync { mapLibreMap ->
        // Set an empty style
        mapLibreMap.setStyle(Style.Builder().fromJson("{}")) { style ->

            // Add a test RasterSource with tile placeholders
            val testSource = RasterSource(
                "radar-source",
                "https://tile.openstreetmap.org/{z}/{x}/{y}.png",
                256
            )

            style.addSource(testSource)

            // Add a RasterLayer using the test source
            style.addLayer(
                RasterLayer("test-layer", "radar-source").apply {
                    setProperties(PropertyFactory.rasterOpacity(0.7f))
                }
            )
        }
    }
})
```

The {z}, {x}, and {y} placeholders are not being replaced, and the raw URL with the placeholders is being requested, resulting in failed tile loading. This issue causes the map to show no tiles, and the following error is logged:
`[HTTP] This request was cancelled (https://tile.openstreetmap.org/%7Bz%7D/%7Bx%7D/%7By%7D.png). This is expected for tiles that were being prefetched but are no longer needed for the map to render.`

This issue does not occur when the RasterSource and RasterLayer are defined in the style JSON directly (for example, using a style JSON file). The issue seems to be specific to programmatically adding the source and layer.

**Comment by andreas-umbricht** [comment]
Currently on version 11.11.0 I am running into the same problem. @saevioapps were you able to resolve this?

I managed to get something to show up using `val rasterSource = RasterSource(sourceName, TileSet("2.1.0", layer.url))`. I found it on SO. I don't know why that one works at all. I don't know why 2.1.0 is needed there, and I don't know why the vararg tiles just plainly accepts my url and does what I want.

This is super confusing..

**Comment by louwers** [comment]
You need to use a TileSet I think. https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-tile-set/index.html?query=class%20TileSet(val%20tilejson:%20String,%20tiles:%20String)

**Comment by andreas-umbricht** [comment]
Yes, that is what I did. With the code I provided in my comment, it totally works. But I cannot figure out what that first parameter of TileSet (tilejson) is about, since I can enter basically an arbitrary string and it does what it should do and the documentation does not explain anything.

**Comment by andreas-umbricht** [comment]
Okay, nevermind I found [this](https://github.com/mapbox/tilejson-spec/blob/master/2.1.0/README.md) that explains it somehow. But still I think the API is somewhat weird.
